install/update_0.0_2.5.php

```diff
@@ -414,8 +415,65 @@ function plugin_formcreator_updateIssue_2_5(Migration $migration) {
+               CONCAT('f_',`fanswer`.`id`)    AS `display_id`,
+               `fanswer`.`id`                 AS `original_id`,
+               'PluginFormcreatorForm_Answer' AS `sub_itemtype`,
+               `f`.`name`                     AS `name`,
+               `fanswer`.`status`             AS `status`,
+               `fanswer`.`request_date`       AS `date_creation`,
+               `fanswer`.`request_date`       AS `date_mod`,
+               `fanswer`.`entities_id`        AS `entities_id`,
+               `fanswer`.`is_recursive`       AS `is_recursive`,
+               `fanswer`.`requester_id`       AS `requester_id`,
+               `fanswer`.`validator_id`       AS `validator_id`,
+               `fanswer`.`comment`            AS `comment`
+            FROM `glpi_plugin_formcreator_forms_answers` AS `fanswer`
+            LEFT JOIN `glpi_plugin_formcreator_forms` AS `f`
+               ON`f`.`id` = `fanswer`.`plugin_formcreator_forms_id`
+            LEFT JOIN `glpi_items_tickets` AS `itic`
+               ON `itic`.`items_id` = `fanswer`.`id`
+               AND `itic`.`itemtype` = 'PluginFormcreatorForm_Answer'
+            WHERE `fanswer`.`is_deleted` = '0'
+            GROUP BY `original_id`
+            HAVING COUNT(`itic`.`tickets_id`) != 1
+            UNION
+            SELECT DISTINCT
+               NULL                          AS `id`,
+               CONCAT('t_',`tic`.`id`)       AS `display_id`,
+               `tic`.`id`                    AS `original_id`,
+               'Ticket'                      AS `sub_itemtype`,
+               `tic`.`name`                  AS `name`,
+               `tic`.`status`                AS `status`,
+               `tic`.`date`                  AS `date_creation`,
+               `tic`.`date_mod`              AS `date_mod`,
+               `tic`.`entities_id`           AS `entities_id`,
+               0                             AS `is_recursive`,
+               `tic`.`users_id_recipient`    AS `requester_id`,
+               0                             AS `validator_id`,
+               `tic`.`content`               AS `comment`
+            FROM `glpi_tickets` AS `tic`
+            LEFT JOIN `glpi_items_tickets` AS `itic`
+               ON `itic`.`tickets_id` = `tic`.`id`
+               AND `itic`.`itemtype` = 'PluginFormcreatorForm_Answer'
+            WHERE `tic`.`is_deleted` = 0
+            GROUP BY `original_id`
+            HAVING COUNT(`itic`.`items_id`) <= 1";
+
+   $countQuery = "SELECT COUNT(*) AS `cpt` FROM ($query) AS `issues`";
+   $result = $DB->query($countQuery);
+   if ($result !== false) {
+      $count = $DB->fetch_assoc($result);
+      $table = PluginFormcreatorIssue::getTable();
+      if (countElementsInTable($table) != $count['cpt']) {
+         if ($DB->query("TRUNCATE `$table`")) {
+            $DB->query("INSERT INTO `$table` SELECT * FROM ($query) as `dt`");
+            $volume = 1;
+         }
+      }
+   }
```

Comment:
the request depends on other tables which may change un later upgrades. Then we must use the sql query of the version 2.5.